identity/aziot-identityd/src/lib.rs

```diff
@@ -390,49 +399,6 @@ impl Api {
         Ok(())
     }
 
-    async fn dps_get_registered_device(
```

Comment:
This is a loop that calls get_operation_status in a loop after the register call has been made. This is how DPS provisioning works and we seem to be removing it. Is this intentional?

Reply:
I've moved the looping logic into the DPS client itself

Reply:
I missed that in my first pass at the giant review :) LGTM